src/docs/arch-eviction.dox

```diff
@@ -1,15 +1,108 @@
+content on the page that is being evicted.
+
+Eviction runs in the background continuously looking for pages when an application approaches the
+maximum cache size.
+
+@section clean_dirty_data Clean vs dirty data
+- Clean refers to data in the cache that is identical to the version stored on disk.
+- Dirty refers to data in the cache that has been modified and needs to be reconciled with the data stored in disk.
+
+There are three types of eviction in WiredTiger:
+- Clean eviction
+- History store eviction
+- Update restore eviction
+
+@subsection clean_eviction_1 Clean eviction
+In the case of clean eviction, there is no dirty content on the page and it is directly removed from the
+memory. After that, the page remains solely on the disk unchanged.
+
+@subsection hs_eviction_1 History store eviction
+For history store eviction, the page is modified but the changes are all committed. Therefore, the page
+should be clean and can be removed from memory after the dirty changes are written to the new disk
+image, which is the data format WiredTiger stores on disk. The process that builds the disk image
+is called reconciliation. In reconciliation, WiredTiger writes the newest committed value of each
+key to the disk image. All the older values of the key are moved to the history store table. In the
+end, reconciliation flushes the new disk image to the data file on the disk and later eviction
+removes the old page from memory.
+The history store table is the internal table for WiredTiger to store historical values for all the
+user-created tables. Each entry in the history store represents a historical value of a key in the
+user-created tables. The pages in the history store table need to be evicted and loaded back to memory
+as well. Different from the user-created tables, all the content on the history store page is committed
+as the historical values it stores must all be committed, and each entry only has one value and a
+deletion marking the expiration of the entry on its update chain. Therefore, history store pages
+should always be clean after reconciliation and can be evicted from memory. Besides, only having
+one committed value for each key in the history store prevents eviction from moving content already
+in the history store recursively to the history store when evicting a history store page because we
+only move the values after the first committed value to the history store.
+
+@subsection update_restore_eviction_1 Update restore eviction
+Update restore eviction largely overlaps with history store eviction. The difference is that there
+are uncommitted changes on the page and WiredTiger cannot write them to the disk image. Instead, besides
+writing the disk image to the disk, eviction also keeps the disk image in memory and restores the
+uncommitted changes to the new disk image.
+Since usually there is only a subset of keys that have uncommitted changes, eviction only needs to
```

Comment:
Though each key is checked for whether there are updates that are not evictable, the information around whether update-restore needs to be performed is decided per reconciled page. So even if a single update (for just one key) on a single reconciled page is not evictable, that whole page needs to be update-restored.

It might be okay to talk about this in less detail, as a short note, and rather get more details around the overall process of eviction (server / candidates / cache management /etc)